frontend/src/modules/Dashboard/Components/Dashboard.tsx

```diff
@@ -38,11 +41,77 @@ export const Dashboard = () => {
       setGroups(res.data.data)
     })
   }, [])
+  // (a,b) = -1 if a before b, 1 if a after b, 0 if equal
+  function sorted(groups: CourseInfo[], menuValue: string) {
+    if (menuValue === 'newly-match-first') {
```

Comment:
Perhaps you could turn this stuff into a series of switch statements to make it more readable?